pkg/cloudprovider/plugins/aws/aws_ec2.go

```diff
@@ -51,27 +51,36 @@ type ec2ResourcesCacheSnapshot struct {
 	vpcPeers      map[string][]string
 }
 
-func newEC2ServiceConfig(accountNamespacedName types.NamespacedName, service awsServiceClientCreateInterface,
-	credentials *awsAccountConfig) (internal.CloudServiceInterface, error) {
-	// create ec2 sdk api client.
-	apiClient, err := service.compute()
-	if err != nil {
-		return nil, fmt.Errorf("error creating ec2 sdk api client for account : %v, err: %v", accountNamespacedName.String(), err)
-	}
+// newEC2ServiceConfig creates an ec2 service configs for each region. It returns a region to service config map for successful creations
+// and multierr for any errors.
+func newEC2ServiceConfig(accountNamespacedName types.NamespacedName, services map[string]awsServiceClientCreateInterface) (
+	map[string]internal.CloudServiceInterface, error) {
+	var retErr error
+	configs := make(map[string]internal.CloudServiceInterface)
+	for region, service := range services {
+		// create ec2 sdk api client.
+		apiClient, err := service.compute()
+		if err != nil {
+			err = fmt.Errorf("error creating ec2 sdk api client for account: %s, region: %s, err: %v", accountNamespacedName.String(),
```

Comment:
Can you try/simulae a scenario where this function returns a multi error (i.e. multiple errors) and check how it would look on the CPA status field?